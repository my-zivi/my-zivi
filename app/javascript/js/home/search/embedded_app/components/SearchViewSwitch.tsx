import * as React from 'preact/compat';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { IconProp } from '@fortawesome/fontawesome-svg-core';

export enum SearchView {
  Map = 'map',
  Tiles = 'tiles',
}

interface Props {
  activeView: SearchView;
  onChange: (view: SearchView) => void;
}

const AvailableTiles: Record<SearchView, { icon: IconProp, title: string }> = {
  [SearchView.Tiles]: {
    icon: ['fas', 'th'] as IconProp,
    title: MyZivi.translations.search.searchView.tiles,
  },
  [SearchView.Map]: {
    icon: ['fas', 'map'] as IconProp,
    title: MyZivi.translations.search.searchView.map,
  },
};

const SearchViewSwitch: React.FunctionComponent<Props> = ({ activeView, onChange }) => (
    <div className="d-flex justify-content-center">
      <div className="search-view-switch">
        {Object.entries(AvailableTiles).map(([key, { icon, title }]) => (
          <SearchViewSwitchButton
            title={title}
            icon={icon}
            isActive={activeView === key}
            onClick={() => onChange(key as SearchView)}
          />
        ))}
      </div>
    </div>
);

const SearchViewSwitchButton: React.FunctionComponent<{
  title: string,
  icon: IconProp,
  onClick: () => void,
  isActive: boolean
}> = ({ title, icon, isActive, onClick }) => (
  <button type="button" className={`search-view-switch-button ${isActive ? 'active' : ''}`} onClick={onClick}>
    <FontAwesomeIcon icon={icon} className="mr-2" />
    {title}
  </button>
);

export default SearchViewSwitch;
