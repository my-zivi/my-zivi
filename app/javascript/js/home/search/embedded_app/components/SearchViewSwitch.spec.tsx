import { mount } from 'enzyme';
import * as React from 'preact/compat';
import SearchViewSwitch, { SearchView } from './SearchViewSwitch';

describe('SearchViewSwitch', () => {
  it('renders correctly', () => {
    expect(
      mount(<SearchViewSwitch activeView={SearchView.Map} onChange={jest.fn()} />).find('.search-view-switch'),
    ).toMatchSnapshot();
  });

  it('calls onChange when user clicks on a button', () => {
    const onChange = jest.fn();
    const wrapper = mount(<SearchViewSwitch activeView={SearchView.Map} onChange={onChange} />);
    wrapper.find('.search-view-switch-button').at(1).simulate('click');
    expect(onChange).toHaveBeenCalledWith(SearchView.Map);
  });

  it('renders the correct button as active', () => {
    const wrapper = mount(<SearchViewSwitch activeView={SearchView.Map} onChange={jest.fn()} />);
    expect(wrapper.find('.search-view-switch-button.active').text()).toEqual(
      globalThis.MyZivi.translations.search.searchView.map,
    );
  });
});
