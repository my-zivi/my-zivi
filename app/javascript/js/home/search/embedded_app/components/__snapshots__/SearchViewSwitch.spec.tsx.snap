// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchViewSwitch renders correctly 1`] = `
<div
  className="search-view-switch"
>
  <SearchViewSwitchButton
    icon={
      Array [
        "fas",
        "th",
      ]
    }
    isActive={false}
    onClick={[Function]}
    title="Raster"
  >
    <button
      className="search-view-switch-button "
      onClick={[Function]}
      type="button"
    >
      <FontAwesomeIcon
        className="mr-2"
        icon={
          Array [
            "fas",
            "th",
          ]
        }
      />
      Raster
    </button>
  </SearchViewSwitchButton>
  <SearchViewSwitchButton
    icon={
      Array [
        "fas",
        "map",
      ]
    }
    isActive={true}
    onClick={[Function]}
    title="Karte"
  >
    <button
      className="search-view-switch-button active"
      onClick={[Function]}
      type="button"
    >
      <FontAwesomeIcon
        className="mr-2"
        icon={
          Array [
            "fas",
            "map",
          ]
        }
      />
      Karte
    </button>
  </SearchViewSwitchButton>
</div>
`;
